lib/vmpooler/api/v1.rb

```diff
@@ -499,6 +507,8 @@ def need_token!
                     arg[tag] = arg[tag].match(filter).captures.join if arg[tag].match(filter)
                   end
```

Comment:
While it's nominally outside the changes in the PR, it tends to be confusing (and often brittle) to modify a hash while it's being iterated.

I would see this block of code as being a good candidate for extraction to a method with an intention-revealing name, perhaps something like:

``` ruby
  when 'tags'
    arg = filter_tags(arg)
    export_tags(params[:hostname], arg)
```

with

``` ruby
  def filter_tags(tags = {})
    return unless Vmpooler::API.settings.config[:tagfilter]
    result = {}
    tags.each_pair do |tag, value|
      next unless filter = Vmpooler::API.settings.config[:tagfilter][tag]
      result[tag] = value.match(filter).captures.join if value.match(filter)
    end
  end

  def export_tags(hostname, tags)
    tags.each_pair do |tag, value|
      backend.hset('vmpooler__vm__' + hostname, 'tag:' + tag, value)
      backend.hset('vmpooler__tag__' + Date.today.to_s, hostname + ':' + tag, value)
  end
```